tools/ci-build

```diff
@@ -218,7 +218,6 @@ case $configure in
   *) error "internal error";;
 esac
 
-$make $jobs coldstart
 $make $jobs core
```

Comment:
This feels very wrong.  If you start with a fresh copy of the sources (not from a previous build), `boot/` is missing a number of files (`boot/ocamlrun` + the stdlib) that `make core` needs.  It's the job of `make coldstart` to populate `boot/`.

More generally, I strongly suggest to do `$make $jobs world.opt` and skip the bootstrap in the main CI test.  Reasons:
* `make world.opt` is what most if not all users do.  In the current `ci-build` we have a build procedure that doesn't match the procedure(s) that the end-user will apply.  With accompanying risk of divergence.
* It is wasteful to test the bootstrap on all configurations every time.  Let's have a specific bootstrap test in the other-configs CI test.

Reply:
But `core` does `coldstart` so it works.

Good point about `world.opt`, but what will users really do? If they go by `README.adoc`, they'll do:
```
make world
make bootstrap [optional]
make opt
make opt.opt
```
and then they are told that they could have done `make world.opt`.

If they go by `README.win32.adoc` they'll do:
```
make world bootstrap opt opt.opt install
```
If they use OPAM, they'll do:
```
make world
make world.opt
```
I see much room for improvement here.

Reply:
> But core does coldstart so it works.

Urgh, I didn't guess that.

I would still very much prefer `make world.opt` or, almost equivalently,
```
make world
make opt
make opt.opt
```
The bootstrap needs to be de-emphasized from README.adoc.  In the very early days, it was the best test we had that the newly-built `ocamlc` worked.  Nowadays we have the test suite as a much more comprehensive test.  Building the .opt compilers is also a pretty good way to exercise a compiler that we just built, although it is `ocamlopt` rather than `ocamlc`.

> I see much room for improvement here.
Agreed.